pandas/indexes/base.py

```diff
@@ -1403,46 +1407,31 @@ def _ensure_compat_append(self, other):
             to_concat.append(other)
 
         for obj in to_concat:
-            if (isinstance(obj, Index) and obj.name != name and
-                    obj.name is not None):
-                name = None
-                break
+            if not isinstance(obj, Index):
+                raise TypeError('all inputs must be Index')
```

Comment:
yes, `_ensure_index` would convert list to Index. @jreback are you saying `Index.append` should accept lists?

On second thought, maybe I misread the `Index.append` docstring. The "Index or list/tuple of indices" can be interpreted as both as "list/tuple of Index objects" (so no guarantee to accept list as Index object) or "list of index labels". 
But actually probably the first interpretation. In that case, I think this PR is OK, but @sinhrks it would maybe be good to list this in the whatsnew as change as well? (or do we regard it as a bug fix?)